### PR TITLE
Clean up test documentation and add tests

### PR DESCRIPTION
**Pull request by TimAbraldes**

### CODE REVIEW

**Comment by TimAbraldes** [comment]
I removed the WIP tag from the title. There's lots more to do (i.e. tons more tests that can be written) but this can land piecemeal

File: engine/index.js [hunk 1]
```diff
@@ -462,3 +462,8 @@ export default {
   buildStatus,
   buildIndex,
 };
+
+const test = {
```
Comment (marco-c) [inline engine/index.js]: You could try using `rewire` to avoid exporting internal functions.

https://github.com/jhnns/rewire

Comment (TimAbraldes) [inline engine/index.js]: Rewire looks super cool! I'll update this to use it.

Comment (TimAbraldes) [inline engine/index.js]: It turns out that rewire doesn't work with babel: https://github.com/jhnns/rewire/issues/62

However there is a babel plugin that might work for this use case: https://github.com/speedskater/babel-plugin-rewire

I'm giving it a shot now

Comment (TimAbraldes) [inline engine/index.js]: The babel plugin doesn't work with babel 6+:
  https://github.com/speedskater/babel-plugin-rewire/issues/71

At this point I'm thinking that exporting the items we want to test via a non-default export makes the most sense.

Comment (marco-c) [inline engine/index.js]: OK, let's file a follow-up to use `rewire` or `babel-plugin-rewire` once it's feasible.

Comment (marco-c) [inline engine/index.js]: I've filed #272.

Comment (TimAbraldes) [inline engine/index.js]: Thanks!

**Pull request merged**